src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -3214,6 +3249,35 @@ int wm_vuldet_oval_xml_parser(OS_XML *xml, XML_NODE node, wm_vuldet_db *parsed_o
                 }
 
             }
+        } else if (dist == FEED_REDHAT && !strcmp(node[i]->element, XML_RPM_ARCHITECTURE)) {
+            if (node[i]->attributes && *node[i]->attributes && node[i]->content && *node[i]->content) {
+                os_calloc(1, sizeof(char *), parsed_oval->info_states->arch);
+                for (j = 0; node[i]->attributes[j]; j++) {
+                    if (!strcmp(node[i]->attributes[j], XML_OPERATION)) {
+                        if (node[i]->values[j] && !strcmp(node[i]->values[j], "equals")) {
+                            os_strdup(node[i]->content, parsed_oval->info_states->arch[0]);
+                        } else if (node[i]->values[j] && !strcmp(node[i]->values[j], "pattern match")) {
+                            char * archs;
+                            char * arch_item;
+                            char * saveptr;
+                            int k = 0;
+                            os_strdup(node[i]->content, archs);
```

Comment:
As per our[ coding style guide](https://github.com/wazuh/wazuh/wiki/Coding-style-guide-%28C%29#initialization), variables must be always initialized. It's good practice and prevents unexpected errors.